1/load schema

WickyNilliams opened the pull request:
### What is the context of this PR?
This feature was mostly complete, so this PR is adding (and establishing best practices) for tests. 

### How to review 
Run test suite (`yarn test`), ensure tests pass

WickyNilliams commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Load Existing Schema (5)](https://trello.com/c/JJ3gUf94/1-load-existing-schema-5)

hamishtaplin commented on src/actions/survey.js:
```diff
@@ -23,6 +29,27 @@ export function loadSurvey(surveyData) {
+  };
+}
+
+export function loadSurveyFailure(error) {
```
should probably rename action creators with the same  structure as the action names (noun first)

WickyNilliams replied:
I followed the advice here in this article: https://medium.com/@kylpo/redux-best-practices-eef55a20cc72

They recommend `NOUN_VERB` for action names, since it gives them namespacing. But then they recommend `verbNoun()` for the action creator, since it's the more natural function name - if you wrote the function with no concern for redux, `loadSurvey` would be how you'd name it.

Seemed to make sense to me. What do you think?

hamishtaplin replied:
Cool, so long as there's a convention.

hamishtaplin commented on src/actions/survey.test.js:
```diff
@@ -0,0 +1,84 @@
+  loadSurveySuccess
+} from "actions/survey";
+
+function actionMatching(type, payload = expect.anything()) {
```
Be good to move these into test utils file 🙃

WickyNilliams replied:
how about `actions/survey`?

WickyNilliams replied:
Did you create a test utils folder somewhere in the end? If so, what was the path?

hamishtaplin replied:
`src/tests/utils`

I put it into `src` because I couldn't get Jest to look outside that directory without using a relative path.

WickyNilliams replied:
Sweet, i'll add them there

hamishtaplin commented on src/actions/survey.test.js:
```diff
@@ -0,0 +1,84 @@
+  return createTextFile(JSON.stringify(obj), name);
+}
+
+describe("actions", () => {
```
maybe "survey actions"?

WickyNilliams replied:
How about `actions/survey`? So it mirrors the path

hamishtaplin replied:
👍

WickyNilliams commented:
@hamishtaplin all points addressed

WickyNilliams commented:
@samiwel i've rebased now, so is good to merge. Up to you if you squash. I've already squashed the noisy commits locally, so the commits should all be atomic
